### PR TITLE
PLAN-781

### PR DESCRIPTION
**Pull request by george-silva**

### CODE REVIEW

File: src/planscape/collaboration/views.py [hunk 1]
```diff
@@ -46,3 +50,20 @@ def post(self, request, format=None):
...
             # this will return a 500
             raise
+
+
+class UserObjectRolesForObject(APIView):
+
+    def get(self, request: Request, target_entity: str, object_pk: int):
+
+        user = request.user
+        if not check_for_permission(user.pk, "add_collaborator"):
```
Comment (pflopez) [inline src/planscape/collaboration/views.py:60]: I think here you should probably use `CollaboratorPermission.can_add` instead, as you are not including the case of the user being the creator

File: src/planscape/collaboration/views.py [hunk 2]
```diff
@@ -46,3 +50,20 @@ def post(self, request, format=None):
...
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        content_type = ContentType.objects.get(model=target_entity)
+        user_object_roles = UserObjectRole.objects.filter(
+            content_type=content_type,
+            object_pk=object_pk,
+        ).exclude(collaborator_id=user.pk)
```
Comment (pflopez) [inline src/planscape/collaboration/views.py:67]: not sure what this means, we are excluding the user_object_roles which you are a collaborator? Seem like we want the oposite

File: src/planscape/collaboration/views.py [hunk 3]
```diff
@@ -46,3 +51,21 @@ def post(self, request, format=None):
...
+
+
+class GetInvitationsForObject(APIView):
+    def get(self, request: Request, target_entity: str, object_pk: int):
+        user = request.user
+        content_type = ContentType.objects.get(model=target_entity)
+        Model = content_type.model_class()
+        instance = Model.objects.get(pk=object_pk)
+
+        if not CollaboratorPermission.can_add(user, instance):
```
Comment (pflopez) [inline src/planscape/collaboration/views.py:63]: ahg sorry, i messed up this is probably can_view, no? This method gets the list of invites, not adding new invites

Comment (pflopez) [reply src/planscape/collaboration/views.py:63]: yes, but we'll check for that on the endpoint that adds, no?

Comment (pflopez) [reply src/planscape/collaboration/views.py:63]: > BUT, he needs to be able to add to see the modal no?
> yes, but we'll check for that on the endpoint that adds, no?

I mean technically no. Just happens that on our role structure there are no roles that only can see the modal, but those actions (viewing the list of invites and adding new invites) can be considered separate things

File: src/planscape/planscape/url_converters.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class ContentTypeURLConverter:
+    # in the future, if we do need to
```
Comment (lastminutediorama) [inline src/planscape/planscape/url_converters.py:2]: :heavy_check_mark:

File: src/planscape/collaboration/views.py [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
+from collaboration.models import UserObjectRole
+from collaboration.permissions import CollaboratorPermission
 from collaboration.serializers import (
     CreateUserObjectRolesSerializer,
     UserObjectRoleSerializer,
 )
 import logging
 
 from collaboration.services import create_invite
+from collaboration.utils import check_for_permission
```
Comment (lastminutediorama) [inline src/planscape/collaboration/views.py:16]: we don't need this anymore, I think?

**Review by lastminutediorama — APPROVED** [review]
lgtm. learning to trust the Django path converter magic

**Review by pflopez — APPROVED** [review]

**Pull request merged**